lisa/wa.py

```diff
@@ -598,16 +609,28 @@ def trace_idle_analysis(trace):
     NAME = 'trace'
     _ARTIFACT_NAME = 'trace-cmd-bin'
     _PURE_GET_JOB_DF = False
+    _FORCE_AUTO = True
 
-    def __init__(self, wa_output, trace_to_df, **kwargs):
+    def __init__(self, wa_output, trace_to_df=None, **kwargs):
```

Comment:
We definitely don't want to do that. I've honestly never seen any code where something is either X or None and does not eventually trip itself, it's quite amazing how consistently we fail at this. The only salvation can come from mypy or simply never doing this (current approach). 

Here it's easy, you can simply make `trace_to_df` default to a function that raises an exception and that's even less code.